scripts/build-local.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+# pylint: disable=invalid-name
+
+from argparse import ArgumentParser
+import copy
+from pathlib import Path
+import re
+import shutil
+import sys
+
+import yaml
+
+NORMAL = '\033[0m'
+RED = '\033[01;31m'
+GREEN = '\033[01;32m'
+
+try:
+    import tuxmake.build
+except ModuleNotFoundError:
+    print(
+        'The tuxmake package is required to use this script but it could not be found!'
+    )
+    print(
+        'Follow the "Installing TuxMake" section of https://tuxmake.org to install it.'
+    )
+    sys.exit(1)
+
+tuxmake_dir = Path(__file__).resolve().parents[1].joinpath('tuxmake')
+default_build_dir = Path(tuxmake_dir, 'build')
+default_output_dir = Path(tuxmake_dir, 'output')
+
+parser = ArgumentParser(
+    description='Build TuxSuite YAML files locally using TuxMake')
+parser.add_argument('-b',
+                    '--build-dir',
+                    default=default_build_dir,
+                    help=f"Build folder (default: {default_build_dir})")
+parser.add_argument('-C',
+                    '--directory',
+                    help='Path to kernel source',
+                    required=True)
+parser.add_argument('-d',
+                    '--debug',
+                    action='store_true',
+                    help='Show debugging messages')
+parser.add_argument('-f',
+                    '--files',
+                    help='TuxSuite YAML files to build',
+                    nargs='+',
+                    required=True)
+parser.add_argument('-j',
+                    '--jobs',
+                    help='Jobs to build (default: build all jobs)',
+                    nargs='+')
+parser.add_argument(
+    '-o',
+    '--output-dir',
+    default=default_output_dir,
+    help=f"Output folder for TuxMake files (default: {default_output_dir})")
+parser.add_argument('-v',
+                    '--verbose',
+                    action='store_true',
+                    help="Show tuxmake's output")
+args = parser.parse_args()
+
+if not (tree := Path(args.directory).resolve()).exists():
+    raise FileNotFoundError(
+        f"Requested tree ('{tree}'), derived from provided directory ('{args.directory}'), could not be found?"
+    )
+
+if shutil.which('podman'):
+    runtime = 'podman'
+elif shutil.which('docker'):
+    runtime = 'docker'
+else:
+    raise RuntimeError(
+        "tuxmake requires either podman or docker for versioned toolchains but neither could be found on your system!"
+    )
+
+# Combine all jobs into one object for easy iteration
+jobs = {}
+for file_str in args.files:
+    with open(file_str, encoding='utf-8') as file:
+        config = yaml.safe_load(file)
+
+    current_jobs = config['jobs']
```

Comment:
`current_jobs` is unused.

Reply:
38b6ca77caa22c260704538979317f7a3cb016b9